### PR TITLE
Check dir existence using os.path.isdir

### PR DESCRIPTION
**Pull request by aptxkid**
Using 'test -d' works fine only on POSIX systems.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @aptxkid has signed the CLA. Thanks for the pull request!

File: test/unit/project_type/test_directory.py [hunk 1]
```diff
@@ -47,8 +47,25 @@ def setUp(self):
...
         directory_env = Directory(project_directory)
         actual_timing_file_path = directory_env.timing_file_path(fake_job_name)
 
         self.assertEqual(actual_timing_file_path, expected_timing_file_path)
+
+    @genty_dataset(
+        (True, False),
+        (False, True),
+    )
+    def test_fetch_project(self, expect_dir_exists, expect_runtime_error):
```
Comment (josephharrington) [inline test/unit/project_type/test_directory.py]: Want to be more specific in the test name about what this is testing? :grin:

File: app/project_type/directory.py [hunk 2]
```diff
@@ -35,9 +35,8 @@ def __init__(self, project_directory, config=None, job_name=None, build_project_
         self._logger.debug('Project directory is {}'.format(project_directory))
 
     def _fetch_project(self):
-        check_command = 'test -d "{}"'.format(self.project_directory)
-        output, exit_code = self.execute_command_in_project(check_command, cwd='/')
-        if exit_code != 0:
+        dir_exists = os.path.isdir(self.project_directory)
```
Comment (josephharrington) [inline app/project_type/directory.py]: Nice. The old way was dumbbbbb.

**Comment by josephharrington** [comment]
Looks good with one minor comment! :+1:

**Comment by aptxkid** [comment]
`test_shutting_down_after_running_a_build_does_not_raise_exception` is a flaky test. Decorating it with `genty_repeat` I can reproduce the failure locally.

**Pull request merged**